datafusion/catalog/src/table.rs

```diff
@@ -55,6 +55,11 @@ pub trait TableProvider: Debug + Sync + Send {
     /// Get a reference to the schema for this table
     fn schema(&self) -> SchemaRef;
 
+    /// Get metadata columns of this table.
+    fn metadata_columns(&self) -> Option<SchemaRef> {
```

Comment:
Can you please document this better -- specifically:
1. A link to the prior art (spark metadata columns)
2. A brief summary of what metadata columns are used for and an example (you can copy the content from the spark docs)